cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -132,20 +132,22 @@ func CreateStaticPodFiles(manifestDir, patchesDir string, cfg *kubeadmapi.Cluste
 // getAPIServerCommand builds the right API server command from the given config object and version
 func getAPIServerCommand(cfg *kubeadmapi.ClusterConfiguration, localAPIEndpoint *kubeadmapi.APIEndpoint) []string {
 	defaultArguments := map[string]string{
-		"advertise-address":               localAPIEndpoint.AdvertiseAddress,
-		"insecure-port":                   "0",
-		"enable-admission-plugins":        "NodeRestriction",
-		"service-cluster-ip-range":        cfg.Networking.ServiceSubnet,
-		"service-account-key-file":        filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPublicKeyName),
-		"client-ca-file":                  filepath.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
-		"tls-cert-file":                   filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerCertName),
-		"tls-private-key-file":            filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKeyName),
-		"kubelet-client-certificate":      filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKubeletClientCertName),
-		"kubelet-client-key":              filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKubeletClientKeyName),
-		"enable-bootstrap-token-auth":     "true",
-		"secure-port":                     fmt.Sprintf("%d", localAPIEndpoint.BindPort),
-		"allow-privileged":                "true",
-		"kubelet-preferred-address-types": "InternalIP,ExternalIP,Hostname",
+		"advertise-address":                localAPIEndpoint.AdvertiseAddress,
+		"insecure-port":                    "0",
+		"enable-admission-plugins":         "NodeRestriction",
+		"service-cluster-ip-range":         cfg.Networking.ServiceSubnet,
+		"service-account-key-file":         filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPublicKeyName),
+		"service-account-signing-key-file": filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPrivateKeyName),
+		"service-account-issuer":           fmt.Sprintf("https://kubernetes.default.svc.%s", cfg.ClusterName),
```

Comment:
would this work?

also this PR seems to do competing changes to: https://github.com/kubernetes/kubernetes/pull/95667
should one of the PRs be closed?

Reply:
#95667 will be focusing on the non flags changes as the flags are required by TokenRequest now not BoundServiceAccountTokenVolume.

Reply:
in https://github.com/kubernetes/kubernetes/pull/95667 we didn't seem to agree whether `https://kubernetes.default.svc` should be used as the default for this.

would `https://kubernetes.default.svc` work out of the box for users that don't specify `--service-account-issuer`?

Reply:
according to https://tools.ietf.org/html/rfc7519#section-4.1.1, i didn't find any reason to not use `https://kubernetes.default.svc` as the default value as it is already used everywhere else in the repo.

@mikedanese WDYT?

Reply:
> The "iss" (issuer) claim identifies the principal that issued the JWT.

Do we consider different clusters to be different security principals? The answer is generally yes but I'm not sure we have a better identifier to shove in here right now. cfg.ClusterName or `https://kubernetes.default.svc` wouldn't be end of the world. I would pick one or the other, and not try to combine them  https://bit.ly/k8s-cluster-id is somewhat relevant.